src/services/map-selection-service/map-selection.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@angular/core';
+
+import { CesiumService } from '../cesium/cesium.service';
+import { PickOptions } from '../map-events-mananger/consts/pickOptions.enum';
+import { CesiumEvent } from '../map-events-mananger/consts/cesium-event.enum';
+import { EventResult, MapEventsManagerService } from '../map-events-mananger/map-events-manager';
+import { EventSelectionInput } from './EventSelectionInput';
+import { UtilsService } from '../../utils/utils.service';
+import { Observable } from 'rxjs/Observable';
+import { PlonterService } from '../plonter/plonter.service';
+import { DisposableObservable } from '../map-events-mananger/disposable-observable';
+import * as _ from 'lodash';
+
+/**
+ * Manages all selection events.
+ * __Notice__: When using multi selection, if registering to the same event more then once, the array of picks will reset.
+ * __usage:__
+ * ```
+ * MapSelectionService.select({event, modifier, priority, entityType, pickOption}).subscribe()
+ * ```
+ * __param:__ {CesiumEvent} event
+ * __param:__ {CesiumEventModifier} modifier
+ * __param:__ priority - the bigger the number the bigger the priority. default : 0.
+ * __param:__ entityType - entity type class that you are interested like (Track). the class must extends AcEntity
+ * __param:__ pickOption - self explained
+ */
+@Injectable()
+export class MapSelectionService {
+    private scene;
+    private multiPickEventMap = new Map<CesiumEvent, Array<{ primitive: any }>>();
+    private triggerPickJson = {
+        'PICK_FIRST': this.triggerPickFirst.bind(this),
+        'PICK_ALL': this.triggerPickAll.bind(this),
+        'MULTI_PICK': this.triggerMultiPick.bind(this)
+    };
+
+
+    constructor(cesiumService: CesiumService,
+                private mapEventsManagerService: MapEventsManagerService,
```

Comment:
This is Map-Selection-Service. As you requested it manage all selection logic, so i moved it out to here from map-events-manager